src/main/java/appeng/fluids/parts/PartFluidStorageBus.java

```diff
@@ -0,0 +1,508 @@
+		if( in != null )
+		{
+			before = in.getAvailableItems( before );
+		}
+
+		this.cached = false;
+		if( fullReset )
+		{
+			this.handlerHash = 0;
+		}
+
+		final IMEInventory<IAEFluidStack> out = this.getInternalHandler();
+
+		if( in != out )
+		{
+			IItemList<IAEFluidStack> after = AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ).createList();
+			if( out != null )
+			{
+				after = out.getAvailableItems( after );
+			}
+			Platform.postListChanges( before, after, this, this.source );
+		}
+	}
+
+	protected void resetCache( final boolean fullReset )
+	{
+		if( this.getHost() == null || this.getHost().getTile() == null || this.getHost().getTile().getWorld() == null || this.getHost().getTile().getWorld().isRemote )
+		{
+			return;
+		}
+
+		if( fullReset )
+		{
+			this.resetCacheLogic = 2;
+		}
+		else
+		{
+			this.resetCacheLogic = 1;
+		}
+
+		try
+		{
+			this.getProxy().getTick().alertDevice( this.getProxy().getNode() );
+		}
+		catch( final GridAccessException e )
+		{
+			// :P
+		}
+	}
+
+	@Override
+	public boolean onPartActivate( final EntityPlayer player, final EnumHand hand, final Vec3d pos )
+	{
+		if( !player.isSneaking() )
+		{
+			if( Platform.isClient() )
+			{
+				return true;
+			}
+			Platform.openGUI( player, this.getHost().getTile(), this.getSide(), GuiBridge.GUI_STORAGEBUS_FLUID );
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public void onChangeInventory( final IItemHandler inv, final int slot, final InvOperation mc, final ItemStack removedStack, final ItemStack newStack )
+	{
+		super.onChangeInventory( inv, slot, mc, removedStack, newStack );
+
+		if( inv == this.config )
+		{
+			this.resetCache( true );
+		}
+	}
+
+	@Override
+	public IItemHandler getInventoryByName( final String name )
+	{
+		if( name.equals( "config" ) )
+		{
+			return this.config;
+		}
+
+		return super.getInventoryByName( name );
+	}
+
+	@Override
+	public void readFromNBT( final NBTTagCompound data )
+	{
+		super.readFromNBT( data );
+		this.config.readFromNBT( data, "config" );
+	}
+
+	@Override
+	public void writeToNBT( final NBTTagCompound data )
+	{
+		super.writeToNBT( data );
+		this.config.writeToNBT( data, "config" );
+	}
+
+	@Override
+	public boolean isValid( final Object verificationToken )
+	{
+		return this.handler == verificationToken;
+	}
+
+	@Override
+	public void postChange( final IBaseMonitor<IAEFluidStack> monitor, final Iterable<IAEFluidStack> change, final IActionSource source )
+	{
+		try
+		{
+			if( this.getProxy().isActive() )
+			{
+				this.getProxy().getStorage().postAlterationOfStoredItems( AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ), change, this.source );
+			}
+		}
+		catch( final GridAccessException e )
+		{
+			// :(
+		}
+	}
+
+	public MEInventoryHandler<IAEFluidStack> getInternalHandler()
+	{
+		if( this.cached )
+		{
+			return this.handler;
+		}
+
+		final boolean wasSleeping = this.monitor == null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "wasSleeping" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)